### PR TITLE
GEF statuses as the actual status instead of uppercase constant

### PR DESCRIPTION
**Pull request by ttbarnes**
## Summary
- Update GEF statuses to use actual status instead of uppercase name. E.g instead of `status: "SUBMITTED"` in the DB, it’s now `status: “Submitted to UKEF”`

## Necessities
- Replace all hard coded GEF statuses with imports/references from constants
- Update any GEF templates that have conditions based on the constants
- Update GEF deal statuses in TFM constants

## Clean up
- Remove GEF `PROGRESS` constants and `STATUS` enums, it’s the same as `STATUS`/`DEAL_STATUS`
- Remove `getStatusLabel` nunjucks filter used in Portal, no longer required

### CODE REVIEW

**Review by IgnatG — COMMENTED** [review]
LGTM!
Just waiting for the tests to finish and then I can approve it

File: gef-ui/templates/includes/application-details/about-exporter.njk [hunk 1]
```diff
@@ -19,7 +19,7 @@
   </div>
 </div>
 
-{% if exporter.status.code === 'NOT_STARTED' and not previewMode %}
+{% if exporter.status.code === 'Not started' and not previewMode %}
```
Comment (abhi-markan) [inline gef-ui/templates/includes/application-details/about-exporter.njk:22]: Can we centralised constants on njk as well to keep it consistent across all the templates, any thoughts?
Could be a future PR just a thought.

File: portal/server/controllers/dashboard/deals.js [hunk 2]
```diff
@@ -85,7 +85,7 @@ exports.gefDeals = async (req, res) => {
   if (isChecker && !isMaker) {
     filters.push({
       field: 'status',
-      value: 'BANK_CHECK',
+      value: 'Ready for Checker\'s approval',
```
Comment (abhi-markan) [inline portal/server/controllers/dashboard/deals.js:88]: Should load value from constant please.

File: portal/server/controllers/dashboard/deals.test.js [hunk 3]
```diff
@@ -123,7 +123,7 @@ describe('controllers/dashboard', () => {
 
       expect(api.gefDeals).toHaveBeenCalledWith(20, 20, [{
         field: 'status',
-        value: 'BANK_CHECK',
+        value: 'Ready for Checker\'s approval',
```
Comment (abhi-markan) [inline portal/server/controllers/dashboard/deals.test.js:126]: Should load value from constant please.

File: portal/server/controllers/dashboard/facilities.js [hunk 4]
```diff
@@ -63,7 +63,7 @@ exports.gefFacilities = async (req, res) => {
   const { user } = req.session;
   const facilityFilters = [];
 
-  if (user.roles.every((role) => role === 'checker')) facilityFilters.push({ field: 'deal.status', value: 'BANK_CHECK', operator: 'eq' });
+  if (user.roles.every((role) => role === 'checker')) facilityFilters.push({ field: 'deal.status', value: 'Ready for Checker\'s approval', operator: 'eq' });
```
Comment (abhi-markan) [inline portal/server/controllers/dashboard/facilities.js:66]: Should load value from constant please.

File: portal/server/controllers/dashboard/facilities.test.js [hunk 5]
```diff
@@ -161,7 +161,7 @@ describe('controllers/facilities', () => {
     it('passes the expected filter for checker', async () => {
       await gefFacilities(checkerReq, res);
 
-      expect(api.gefFacilities).toHaveBeenCalledWith(20, 20, [{ field: 'deal.status', operator: 'eq', value: 'BANK_CHECK' }], 'mock-token');
+      expect(api.gefFacilities).toHaveBeenCalledWith(20, 20, [{ field: 'deal.status', operator: 'eq', value: 'Ready for Checker\'s approval' }], 'mock-token');
```
Comment (abhi-markan) [inline portal/server/controllers/dashboard/facilities.test.js:164]: Should load value from constant please.

File: trade-finance-manager-api/src/v1/__mocks__/mock-gef-deal.js [hunk 6]
```diff
@@ -81,7 +82,7 @@ const MOCK_GEF_DEAL = {
   ],
   facilities: MOCK_CASH_CONTINGENT_FACILITIES,
   eligibility: {
-    status: 'COMPLETED',
+    status: 'Completed',
```
Comment (abhi-markan) [inline trade-finance-manager-api/src/v1/__mocks__/mock-gef-deal.js:85]: Should load value from constant please.

**Review by IgnatG — APPROVED** [review]

**Review by abhi-markan — CHANGES_REQUESTED** [review]
Have left comments and a question regarding feasibility?

**Comment by ttbarnes** [comment]
Thanks @abhi-markan the one's that aren't using constants, didn't have constants in place, i'll add constants.

File: gef-ui/templates/includes/application-details/about-exporter.njk (hunk 1 above)
Comment (ttbarnes) [reply gef-ui/templates/includes/application-details/about-exporter.njk:22]: future PR

**Pull request merged**